orttraining/orttraining/test/gradient/gradient_ops_test.cc

```diff
@@ -1207,6 +1207,113 @@ TEST(GradientCheckerTest, SparseSoftmaxCrossEntropyGrad) {
   TestSparseSoftmaxCrossEntropyGrad({2, 3, 2}, "sum");
 }
 
+void TestSoftmaxCrossEntropyLossGrad(const TensorShape& index_shape,  //label_shape
+                                     const std::string& reduction,
+                                     int64_t ignore_index = 0,
+                                     int64_t D = 2 /* num_class*/) {
+  float max_error;
+  GradientChecker<float, float, float> gradient_checker;
+  OpDef op_def{"SoftmaxCrossEntropyLoss", kOnnxDomain, 12};
+  std::function<float(float)> transformer_index = [D](float x) { return std::fmod(std::fabs(x) * 5.0f, D * 1.0f); };
```

Comment:
I just realized one thing. 
the random generated index will never have -1 as the value. so the ignore_index can is not covered in this UT. 
We might to explicitly provide the index input data in this case. We have examples in this file

Reply:
in bound ignore_index is covered but for out of bounds you are correct, I can add code to explicitly add out of bounds indices. 

---
In reply to: [409271676](https://github.com/microsoft/onnxruntime/pull/3465#discussion_r409271676) [](ancestors = 409271676)